MeDriAnchor/MeDriAnchorETLRun.ps1

```diff
@@ -14,1024 +9,1003 @@ param
+            $SqlCmdEventAlert.Connection = $sqlConnectionMeDriAnchorDB;
+            $SqlCmdEventAlert.ExecuteNonQuery() >> $null;
+
+            # throw a fatal error if we can't connect to do anything
+            $errorMsg = "Error running the sspDropDWHObjectView procedure. Error: " + $_.Exception.ToString();
+	        throw $errorMsg;
+        }
+
+        try
+        {
+            # set numeric roundabort off
+            $SqlCmdMaintNR = New-Object System.Data.SqlClient.SqlCommand;
+            $SqlCmdMaintNR.CommandText = "SET NUMERIC_ROUNDABORT OFF;";
+            $SqlCmdMaintNR.Connection = $sqlConnectionMeDriAnchorDB;
+            $SqlCmdMaintNR.ExecuteScalar();
+
+            $SqlCmdMaint2 = New-Object System.Data.SqlClient.SqlCommand("[MeDriAnchor].[sspCreateDWHObjectView]", $sqlConnectionMeDriAnchorDB);
+            $SqlCmdMaint2.CommandType = [System.Data.CommandType]'StoredProcedure'
+            $SqlCmdMaint2.Parameters.Add("@Environment_ID", [System.Data.SqlDbType]"SmallInt") >> $null;
+            $SqlCmdMaint2.Parameters["@Environment_ID"].Value = [Int32]$environmentid;
+            $SqlCmdMaint2.Parameters.Add("@Debug", [System.Data.SqlDbType]"Bit") >> $null; 
+            $SqlCmdMaint2.Parameters["@Debug"].Value = [Byte]$debug;
+            $SqlCmdMaint2.CommandTimeout = 600;
+            $result = $SqlCmdMaint2.ExecuteScalar();
+
+            $SqlCmdEventAlert.CommandText = "INSERT INTO [MeDriAnchor].[EventAlerts]([Batch_ID], [SeverityID], [AlertMessage]) VALUES (" + $batchid.ToString() + ", 1, 'Created the DWH object view');";
+            $SqlCmdEventAlert.Connection = $sqlConnectionMeDriAnchorDB;
+            $SqlCmdEventAlert.ExecuteNonQuery() >> $null;
+
+        }
+        catch
+        {
+            $SqlCmdEventAlert.CommandText = "INSERT INTO [MeDriAnchor].[EventAlerts]([Batch_ID], [SeverityID], [AlertMessage]) VALUES (" + $batchid.ToString() + ", 5, 'Failed to create the DWH object view');";
+            $SqlCmdEventAlert.Connection = $sqlConnectionMeDriAnchorDB;
+            $SqlCmdEventAlert.ExecuteNonQuery() >> $null;
+
+            # throw a fatal error if we can't connect to do anything
+            $errorMsg = "Error running the sspCreateDWHObjectView procedure. Error: " + $_.Exception.ToString()
+	        throw $errorMsg;
+        }
+
+        #########################################################################################################################
+        ## STEP 5.6: GENERATING DWH SYNONYMS
+        #########################################################################################################################
+
+        write-host "Generating DWH synonyms...." -ForegroundColor Yellow;
 
         try
         {
-            $SqlCmdMaint3 = New-Object System.Data.SqlClient.SqlCommand("[MeDriAnchor].[sspGenerateSourceSynonyms]", $sqlConnectionMeDriAnchorDB);
```

Comment:
This is a pain in the arse!!!!!

I will have lost any fixes you did when I had to compare to my much-changed version that I worked on since mid-December.

I have no idea how to put these back in now. The whole version that I checked in runs fine locally, both with a single DWH and three different environment DWH databases.

I have no idea how to correct whilst I have a branch merge pending?

J

From: Stephanie Locke [mailto:notifications@github.com]
Sent: 06 January 2015 10:34
To: stephlocke/MeDriAnchor
Cc: James Skipwith
Subject: Re: [MeDriAnchor] JS - Version 1.1 with multi-environments now fully supported in seperate... (#4)


In MeDriAnchor/MeDriAnchorETLRun.ps1<https://github.com/stephlocke/MeDriAnchor/pull/4#discussion-diff-22517257>:

>

>          try

>          {

> -            $SqlCmdMaint3 = New-Object System.Data.SqlClient.SqlCommand("[MeDriAnchor].[sspGenerateSourceSynonyms]", $sqlConnectionMeDriAnchorDB);

Looks like you also lost this fix of #3<https://github.com/stephlocke/MeDriAnchor/issues/3>

—
Reply to this email directly or view it on GitHub<https://github.com/stephlocke/MeDriAnchor/pull/4/files#r22517257>.